### ISSUE TITLE
[javascript] match inside function second argument

### ISSUE THREAD

**Issue by jonny64** [returntocorp/sgrep-rules#2476]
**Is your feature request related to a problem? Please describe.**
```
let filter = {key: USER_INPUT}
let data = {... ...}
add_vocabularies (data, {voc_some: {filter}, voc_other: {}, voc_etc: {}})
```
how do I match USER_INPUT in second argument?

**Describe alternatives you've considered**
this gives false positive even if USER_INPUT goes to data (which is OK)
```
    languages:
      - js
    message: this.rq.$X injected in db.add_vocabularies without sanitizing!
    mode: taint
    pattern-sources:
      - pattern: USER_INPUT
    pattern-sinks:
      - pattern: add_vocabularies($DATA, <... {..., filter:$X, ...} ...>)
```

**Use case**
integrating semgrep in workflow, finding meaningful results

**Describe the solution you'd like**
no false positive, match inside function second argument

** Some more **
another question, how do I match this.db when an alias was created
```
const db = this.db
db.add_vocabularies (...)
```

**Comment by r2c-demo** [comment]
This issue is synced in Linear at https://linear.app/r2c/issue/PA-1989/[javascript]-match-inside-function-second-argument. _Note_: this link is for r2c use only and is not accessible publicly.

**Comment by colleend** [comment]
Hi! Thanks for filing this issue -- for matching the second argument, you can do something like this: https://semgrep.dev/s/wjZN where we focus on the metavariable representing the second argument to narrow down the matches. 

For matching `this.db`, you can do something similar to:
```
$DB = this.db
...
$DB.add_vocabularies(...)
```
where the metavariable abstracts away the variable name. Also feel free to join our community slack at r2c-community.slack.com, as that is where we respond faster to rule questions!

**Comment by jonny64** [comment]
what if I want to capture both:
* `const {db, rq, conf} = this` 
* `this.db`

**Comment by jonny64** [comment]
tried https://semgrep.dev/docs/experiments/symbolic-propagation
looks like 
* `let db = this.db` is captured
* `let {db} = this` missed

**Comment by colleend** [comment]
would you mind submitting a semgrep.dev editor link on what you want to capture and what you have so far?

**Comment by LewisArdern** [comment]
@jonny64 is it this you are looking for?

https://semgrep.dev/s/rrEP

In general, questions like this should be asked over at https://r2c.dev/slack so it can be seen by the larger community and we can respond easier for rule issues in how you write a pattern.

**Comment by jonny64** [comment]
https://semgrep.dev/s/jonny64:second-arg-match-copy

**Comment by LewisArdern** [comment]
Hi @jonny64 this is probably what you are looking for. https://semgrep.dev/s/OAGp

In future for questions about writing rules please use https://r2c.dev/slack its how our support team can better assist you!

**Comment by jonny64** [comment]
Thanks!!
